http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/SslCloseCompletionEventHandler.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.nio.netty.internal;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.handler.ssl.SslCloseCompletionEvent;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+
+/**
+ * Handles {@link SslCloseCompletionEvent}s that are sent whenever an SSL channel
+ * goes inactive. This most commonly occurs on a tls_close sent by the server. Channels
+ * in this state can't be reused so they must be closed.
+ */
+@SdkInternalApi
+public class SslCloseCompletionEventHandler extends ChannelInboundHandlerAdapter {
```

Comment:
minor, might be worth creating a singleton across channels to reduce GC since this handler is stateless.

nit, `final` on this class